src/rockstor/fs/btrfs.py

```diff
@@ -715,7 +798,18 @@ def qgroup_create(pool):
 
 
 def qgroup_destroy(qid, mnt_pt):
-    o, e, rc = run_command([BTRFS, 'qgroup', 'show', mnt_pt])
+    cmd = [BTRFS, 'qgroup', 'show', mnt_pt]
+    try:
+        o, e, rc = run_command(cmd, log=True)
+    except CommandException as e:
+        # we may have quotas disabled so catch and deal.
+        emsg = "ERROR: can't list qgroups: quotas not enabled"
+        if e.rc == 1 and e.err[0] == emsg:
```

Comment:
Just a minor concern. Do we need to check e.rc == 1 here? I don't know if btrfs-progs has standardized error codes, so my concern is if the message stays the same in the future where as code is changed for some reason. I'd think that the message is reliable and reliable enough.

Reply:
Agreed. We could just check the message. I was just catching what I saw and checking both to be thorough/precise but as you say it may just trip us up later.